### PR TITLE
Move to testing on solr 6 and use solr_wrapper for Solr

### PR DESCRIPTION
**Pull request by jcoyne**
Do we need to add directions for how to create a solr collection for development, or is the single collection (how we do it in Blacklight) a paradigm that's going to work?

### CODE REVIEW

**Comment by ndushay** [comment]
@jmartin-sul you might want to look at this PR, even if it gets merged, just to make sure you don't want to tweak anything.

**Comment by mejackreed** [comment]
👏

**Comment by atz** [comment]
I like getting Solr 6.  Some reservations though.

Obviously Argo is concerned more with indexing logic and index writes than a read-only app like Blacklight.  Single core makes perfect sense for an app that only ever cares about solr read operations.  But if you want to test two different index configurations/schemas, our dev practices should support that.  

It seems like a major concession to default to nuking a dev's entire index every time they want to run tests, since it basically precludes working on the schema and being able to run the same code/tests against both solr configurations.  Samvera stack as a whole has kept itself painfully detached from using or configuring solr expertly.  This kinda cements that in place in a way I consider unproductive.  

Maybe Argo isn't where we would decide to refine our solr schema (though it was previously), but it seems as likely an app as any.  It is more practical to do it here than in a higher level of abstraction (e.g. an engine/gem).  Argo's index is **the** main index of SDR.  When we test something like a major updates, we have performed costly operations like reindexing hundreds of thousands of objects under a new schema.  The precursor to that is having revisions to the schema.  Making that process dumber or more cumbersome increases the risk of incurring that cost multiple times.  

TL;DR: Not going to block this, but not going to approve it either.
